File: 3752-lexicographically-smallest-negated-permutation-that-sums-to-target/3752-lexicographically-smallest-negated-permutation-that-sums-to-target.java
class Solution {
    public int[] lexSmallestNegatedPerm(int n, long target) {
        long maxi = (long) n * (n + 1) / 2;
        long mini = - maxi;
        if(target < mini || target > maxi ||(maxi - target) % 2 != 0) return new int[0];
        long diff = (maxi - target) / 2;
        int[] ans = new int[n];
        for(int i = n ; i >= 1 ; i--){
            if(diff >= i){
                ans[i - 1] = -i;
                diff -= i;
            }else{
                ans[i - 1] = i;
            }
        }
        Arrays.sort(ans);
        return ans;
    }
}

File: 3752-lexicographically-smallest-negated-permutation-that-sums-to-target/3752-lexicographically-smallest-negated-permutation-that-sums-to-target.cpp
class Solution {
public:
    vector<int> lexSmallestNegatedPerm(int n, long long target) {
        long long maxsum = (long long)n * (n + 1)/2;
        long long minsum = - maxsum;
        
        long long diff =( maxsum - target) / 2;
        if(target < minsum || target > maxsum || (maxsum - target ) % 2 != 0) return {};
        vector<int> ans(n+1 , false);
        for(int i = n ; i >= 1 ; i--){
            if(diff >= i){
                diff -= i;
                ans[i] = true;
            }
        }
        vector<int> res;
        for(int i = n ; i >= 1 ; i--){
            if(ans[i] == true) res.push_back(-i);
        }
        for(int i = 1 ; i <= n ; i++){
            if(ans[i] == false) res.push_back(i);
        }
        return res ;
    }
};